smart-open version 5.0.0 breaks serialization tests

The latest release of smart-open is causing some serialization tests to fail. We should investigate to determine if we can make updates that will work with both newer and older versions of smart-open or if we need to update the requirements permanently.

Related to featuretools issue ([1373](https://github.com/alteryx/featuretools/issues/1373))

```
=========================== short test summary info ============================
FAILED woodwork/tests/accessor/test_serialization.py::test_s3_test_profile[sample_df_pandas]
FAILED woodwork/tests/accessor/test_serialization.py::test_deserialize_s3_csv
```
#### Stack Trace

<details>

```
____________________ test_s3_test_profile[sample_df_pandas] ____________________
[gw1] linux -- Python 3.7.10 /opt/hostedtoolcache/Python/3.7.10/x64/bin/python

sample_df =    id         full_name                   email  ... age  signup_date is_registered
0   0      Mr. John Doe  john.smit...         True
3   3  Ms. Paige Turner        junk@example.com  ...  57   2020-09-01          True

[4 rows x 7 columns]
s3_client = s3.ServiceResource(), s3_bucket = s3.Bucket(name='test-bucket')
setup_test_profile = None

    def test_s3_test_profile(sample_df, s3_client, s3_bucket, setup_test_profile):
        xfail_tmp_disappears(sample_df)
        sample_df.ww.init()
>       sample_df.ww.to_csv(TEST_S3_URL, encoding='utf-8', engine='python', profile_name='test')

woodwork/tests/accessor/test_serialization.py:405: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
woodwork/table_accessor.py:491: in to_csv
    compression=compression, profile_name=profile_name)
woodwork/serialize.py:91: in write_woodwork_table
    use_smartopen(file_path, path, read=False, transport_params=transport_params)
woodwork/s3_utils.py:14: in use_smartopen
    with open(path, 'wb', transport_params=transport_params) as fout:
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/smart_open_lib.py:221: in open
    binary = _open_binary_stream(uri, binary_mode, transport_params)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/smart_open_lib.py:383: in _open_binary_stream
    fobj = submodule.open_uri(uri, mode, transport_params)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:196: in open_uri
    return open(parsed_uri['bucket_id'], parsed_uri['key_id'], mode, **kwargs)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:280: in open
    writebuffer=writebuffer,
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:734: in __init__
    _initialize_boto3(self, client, client_kwargs, bucket, key)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:500: in _initialize_boto3
    client = boto3.client('s3', **init_kwargs)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/boto3/__init__.py:93: in client
    return _get_default_session().client(*args, **kwargs)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/boto3/session.py:263: in client
    aws_session_token=aws_session_token, config=config)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/session.py:797: in create_client
    region_name = self._resolve_region_name(region_name, config)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/session.py:851: in _resolve_region_name
    region_name = self.get_config_variable('region')
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/session.py:241: in get_config_variable
    logical_name)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/configprovider.py:313: in get_config_variable
    return provider.provide()
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/configprovider.py:410: in provide
    value = provider.provide()
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/configprovider.py:471: in provide
    scoped_config = self._session.get_scoped_config()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <botocore.session.Session object at 0x7f0e3ccabc90>

    def get_scoped_config(self):
        """
        Returns the config values from the config file scoped to the current
        profile.
    
        The configuration data is loaded **only** from the config file.
        It does not resolve variables based on different locations
        (e.g. first from the session instance, then from environment
        variables, then from the config file).  If you want this lookup
        behavior, use the ``get_config_variable`` method instead.
    
        Note that this configuration is specific to a single profile (the
        ``profile`` session variable).
    
        If the ``profile`` session variable is set and the profile does
        not exist in the config file, a ``ProfileNotFound`` exception
        will be raised.
    
        :raises: ConfigNotFound, ConfigParseError, ProfileNotFound
        :rtype: dict
    
        """
        profile_name = self.get_config_variable('profile')
        profile_map = self._build_profile_map()
        # If a profile is not explicitly set return the default
        # profile config or an empty config dict if we don't have
        # a default profile.
        if profile_name is None:
            return profile_map.get('default', {})
        elif profile_name not in profile_map:
            # Otherwise if they specified a profile, it has to
            # exist (even if it's the default profile) otherwise
            # we complain.
>           raise ProfileNotFound(profile=profile_name)
E           botocore.exceptions.ProfileNotFound: The config profile (test) could not be found

/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/session.py:340: ProfileNotFound
------------------------------ Captured log call -------------------------------
WARNING  smart_open.utils:utils.py:69 ignoring unsupported keyword arguments: ['session']
___________________________ test_deserialize_s3_csv ____________________________
[gw1] linux -- Python 3.7.10 /opt/hostedtoolcache/Python/3.7.10/x64/bin/python

sample_df_pandas =    id         full_name                   email  ... age  signup_date is_registered
0   0      Mr. John Doe  john.smit...         True
3   3  Ms. Paige Turner        junk@example.com  ...  57   2020-09-01          True

[4 rows x 7 columns]

    def test_deserialize_s3_csv(sample_df_pandas):
        sample_df_pandas.ww.init(index='id')
>       deserialized_df = deserialize.read_woodwork_table(S3_URL)

woodwork/tests/accessor/test_serialization.py:451: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
woodwork/deserialize.py:150: in read_woodwork_table
    use_smartopen(file_path, path, transport_params)
woodwork/s3_utils.py:9: in use_smartopen
    with open(path, "rb", transport_params=transport_params) as fin:
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/smart_open_lib.py:221: in open
    binary = _open_binary_stream(uri, binary_mode, transport_params)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/smart_open_lib.py:383: in _open_binary_stream
    fobj = submodule.open_uri(uri, mode, transport_params)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:196: in open_uri
    return open(parsed_uri['bucket_id'], parsed_uri['key_id'], mode, **kwargs)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:270: in open
    client_kwargs=client_kwargs,
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:546: in __init__
    self.seek(0)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:638: in seek
    self._current_pos = self._raw_reader.seek(offset, whence)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:389: in seek
    self._open_body(start, stop)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:415: in _open_body
    range_string,
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/smart_open/s3.py:302: in _get
    return client.get_object(Bucket=bucket, Key=key, Range=range_string)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/client.py:357: in _api_call
    return self._make_api_call(operation_name, kwargs)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/client.py:663: in _make_api_call
    operation_model, request_dict, request_context)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/client.py:682: in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/endpoint.py:102: in make_request
    return self._send_request(request_dict, operation_model)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/endpoint.py:132: in _send_request
    request = self.create_request(request_dict, operation_model)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/endpoint.py:116: in create_request
    operation_name=operation_model.name)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/hooks.py:356: in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/hooks.py:228: in emit
    return self._emit(event_name, kwargs)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/hooks.py:211: in _emit
    response = handler(**kwargs)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/signers.py:90: in handler
    return self.sign(operation_name, request)
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/signers.py:162: in sign
    auth.add_auth(request)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <botocore.auth.S3SigV4Auth object at 0x7f0e3d585790>
request = <botocore.awsrequest.AWSRequest object at 0x7f0e3e39c1d0>

    def add_auth(self, request):
        if self.credentials is None:
>           raise NoCredentialsError
E           botocore.exceptions.NoCredentialsError: Unable to locate credentials

/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/botocore/auth.py:357: NoCredentialsError
------------------------------ Captured log call -------------------------------
WARNING  smart_open.utils:utils.py:69 ignoring unsupported keyword arguments: ['resource_kwargs']
=============================== warnings summary ===============================
woodwork/tests/accessor/test_statistics.py::test_describe_accessor_method[describe_df_pandas]
woodwork/tests/accessor/test_statistics.py::test_mutual_info_returns_empty_df_properly[sample_df_pandas]
  /home/runner/work/woodwork/woodwork/unpacked_sdist/woodwork/table_accessor.py:134: SettingWithCopyWarning: 
  A value is trying to be set on a copy of a slice from a DataFrame.
  Try using .loc[row_indexer,col_indexer] = value instead
  
  See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
    self._dataframe[name] = updated_series

woodwork/tests/accessor/test_table_accessor.py::test_get_invalid_schema_message_dtype_mismatch[sample_df_pandas]
  /home/runner/work/woodwork/woodwork/unpacked_sdist/woodwork/table_accessor.py:584: TypingInfoMismatchWarning: Operation performed by astype has invalidated the Woodwork typing information:
   dtype mismatch for column id between DataFrame dtype, int64, and Integer dtype, Int64.
   Please initialize Woodwork with DataFrame.ww.init
    TypingInfoMismatchWarning)

woodwork/tests/accessor/test_table_accessor.py::test_get_invalid_schema_message_dtype_mismatch[sample_df_pandas]
  /home/runner/work/woodwork/woodwork/unpacked_sdist/woodwork/table_accessor.py:584: TypingInfoMismatchWarning: Operation performed by astype has invalidated the Woodwork typing information:
   dtype mismatch for column is_registered between DataFrame dtype, bool, and Boolean dtype, boolean.
   Please initialize Woodwork with DataFrame.ww.init
    TypingInfoMismatchWarning)

woodwork/tests/accessor/test_table_accessor.py::test_get_invalid_schema_message_dtype_mismatch[sample_df_pandas]
  /home/runner/work/woodwork/woodwork/unpacked_sdist/woodwork/table_accessor.py:584: TypingInfoMismatchWarning: Operation performed by astype has invalidated the Woodwork typing information:
   dtype mismatch for column full_name between DataFrame dtype, object, and NaturalLanguage dtype, string.
   Please initialize Woodwork with DataFrame.ww.init
    TypingInfoMismatchWarning)

woodwork/tests/accessor/test_table_accessor.py::test_get_invalid_schema_message_dtype_mismatch[sample_df_pandas]
  /home/runner/work/woodwork/woodwork/unpacked_sdist/woodwork/table_accessor.py:584: TypingInfoMismatchWarning: Operation performed by astype has invalidated the Woodwork typing information:
   dtype mismatch for column age between DataFrame dtype, string, and Categorical dtype, category.
   Please initialize Woodwork with DataFrame.ww.init
    TypingInfoMismatchWarning)
```

--- a/docs/source/install.ipynb
+++ b/docs/source/install.ipynb
@@ -69,10 +69,10 @@
     "| Dependency        | Min Version | Notes                                  |\n",
     "|-------------------|-------------|----------------------------------------|\n",
     "| boto3             | 1.10.45     | Required to read/write to URLs and S3  |\n",
-    "| smart_open        | 1.8.4       | Required to read/write to URLs and S3  |\n",
+    "| smart_open        | 5.0.0       | Required to read/write to URLs and S3  |\n",
     "| pyarrow           | 2.0.0       | Required to serialize to parquet       |\n",
     "| dask[distributed] | 2.30.0      | Required to use with Dask DataFrames   |\n",
-    "| koalas            | 1.4.0       | Required to use with Koalas DataFrames |\n",
+    "| koalas            | 1.7.0       | Required to use with Koalas DataFrames |\n",
     "| pyspark           | 3.0.0       | Required to use with Koalas DataFrames |\n",
     "\n",
     "\n",
@@ -99,7 +99,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.6.8"
+   "version": "3.8.5"
   }
  },
  "nbformat": 4,

--- a/docs/source/release_notes.rst
+++ b/docs/source/release_notes.rst
@@ -9,7 +9,7 @@ Release Notes
     * Changes
         * Rename ``FullName`` logical type to ``PersonFullName`` (:pr:`740`)
         * Rename ``ZIPCode`` logical type to ``PostalCode`` (:pr:`741`)
-        * Restrict smart-open version to <5.0.0 (:pr:`750`)
+        * Fix issue with smart-open version 5.0.0 (:pr:`750`)
     * Documentation Changes
         * Update Pygments version requirement (:pr:`751`)
     * Testing Changes

--- a/docs/source/release_notes.rst
+++ b/docs/source/release_notes.rst
@@ -9,7 +9,7 @@ Release Notes
     * Changes
         * Rename ``FullName`` logical type to ``PersonFullName`` (:pr:`740`)
         * Rename ``ZIPCode`` logical type to ``PostalCode`` (:pr:`741`)
-        * Fix issue with smart-open version 5.0.0 (:pr:`750`)
+        * Fix issue with smart-open version 5.0.0 (:pr:`750`, :pr:`758`)
     * Documentation Changes
         * Update Pygments version requirement (:pr:`751`)
     * Testing Changes

--- a/test-requirements.txt
+++ b/test-requirements.txt
@@ -4,5 +4,5 @@ pytest-cov==2.10.1
 pytest-xdist==2.1.0
 boto3>=1.10.45
 moto[all]>=1.3.14
-smart-open>=1.8.4,<5.0.0
+smart-open>=5.0.0
 pyarrow>=2.0.0

--- a/woodwork/s3_utils.py
+++ b/woodwork/s3_utils.py
@@ -21,11 +21,12 @@ def get_transport_params(profile_name):
     Config = import_or_raise("botocore.config", BOTOCORE_ERR_MSG).Config
 
     if isinstance(profile_name, str):
-        transport_params = {'session': boto3.Session(profile_name=profile_name)}
+        session = boto3.Session(profile_name=profile_name)
+        transport_params = {'client': session.client('s3')}
     elif profile_name is False or boto3.Session().get_credentials() is None:
-        transport_params = {
-            'resource_kwargs': {'config': Config(signature_version=UNSIGNED)}
-        }
+        session = boto3.Session()
+        client = session.client('s3', config=Config(signature_version=UNSIGNED))
+        transport_params = {'client': client}
     else:
         transport_params = None
     return transport_params
@@ -48,7 +49,7 @@ BOTOCORE_ERR_MSG = (
 SMART_OPEN_ERR_MSG = (
     "The smart_open library is required to read and write from URLs and S3.\n"
     "Install via pip:\n"
-    "    pip install smart-open\n"
+    "    pip install 'smart-open>=5.0.0'\n"
     "Install via conda:\n"
-    "    conda install smart_open"
+    "    conda install 'smart-open>=5.0.0'"
 )

